File: 0704.BinarySearch/BinarySearch.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        head, tail = -1, len(nums)
        middle = 0
        while tail - head > 1:
            middle = (tail+head)//2
            if nums[middle] == target:
                return middle
            elif nums[middle] > target:
                tail = middle
            else:
                head = middle
        return -1

File: 0704.BinarySearch/BinarySearch.go
func search(nums []int, target int) int {
    head, tail := -1, len(nums)
    var middle int
    for tail - head > 1 {
        middle = int((tail+head)/2)
        if nums[middle] == target {
            return middle
        } else if nums[middle] < target {
            head = middle
        } else {
            tail = middle
        }
    }
    return -1
}